notebooks/Rules.py

```
@@ -0,0 +1,94 @@
+        for _, row in self.df.iterrows():
+            dt = row['Date_Time']
+            recent = self.df[(self.df['Date_Time'] <= dt) & (self.df['Date_Time'] >= dt - timedelta(minutes=max_minutes))]
+            if len(recent) > max_transactions:
+                flagged.append(row['Transaction_Id'])
+        return flagged
+
+    def rule2_high_risk_cities(self, high_risk=['Sindh', 'Balochistan']):
+        flagged = self.df[self.df['Dest_State'].isin(high_risk)]
+        return flagged['Transaction_Id'].tolist()
+
+    def rule3_high_risk_region_and_time(self, region='Balochistan', times=['03:00:00', '04:00:00']):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)